bin/github-review.sh

```diff
@@ -45,53 +41,130 @@ if ! [[ $1 =~ ^[0-9]+$ ]]; then
     exit 1
 fi
 pr=$1
-if ! [[ $2 =~ ^[0-9]+$ ]]; then
-    echo "ERROR: $2 is not a valid Chain ID number" 1>&2
+prChainID="$(gh pr view $pr | sed -nE 's/.*Chain_ID: ([0-9]+).*/\1/p')"
+if ! [[ $prChainID =~ ^[0-9]+$ ]]; then
+    echo "ERROR: $prChainID is not a valid Chain ID number" 1>&2
     usage
     exit 1
 fi
-rpc=$3
+chainlistURL="https://chainlist.org/chain/$prChainID"
+if [[ "$(curl -s "$chainlistURL")" == 'nope' ]]; then
```

Comment:
This kind of check is very brittle - if Chainlist changes how they format errors, it will break.

I recommend:

```suggestion
chainlistURL="https://chainlist.org/chain/$prChainID"
if ! curl --fail -s -o /dev/null "$chainlistURL"; then
```

This uses `--fail` so `curl` exist with a non-0 error code if there is an HTTP error and doesn't rely on how Chainlist present HTTP errors (which is non-standard). We also send the output to `/dev/null` because we don't care about it.

Note that, depending on the chainlist page, it could also run into the "too long" errors you had with the cast command.

Reply:
Nice suggestion, updated. TIL about `--fail`.